cpp/ql/src/Likely Bugs/Memory Management/ImproperNullTermination.ql

```diff
@@ -23,6 +23,10 @@ DeclStmt declWithNoInit(LocalVariable v) {
   not exists(v.getInitializer())
 }
 
+/**
+ * Flow from a buffer that is not not null terminated to a sink that requires
```

Comment:
Yeah, I think it would be valuable to compare the current set of results to a version that uses `semmle.code.cpp.dataflow.DataFlow`, `semmle.code.cpp.ir.dataflow.DataFlow`, or event `semmle.code.cpp.ir.dataflow.TaintTracking`.

Reply:
I just had a try at both.  AST dataflow produces a lot of false positives, as best I can tell because it doesn't really know when a pointed buffer might be overwritten.  e.g.
```
char buffer1[1024];
char buffer2[1024];

strcpy(buffer2, "content");
strcpy(buffer1, buffer2); // <-- I get a false positive on buffer2 here as though the line above didn't exist.
```
The IR DataFlow might do better at that but fails at the source - in this and most other cases an uninitialized buffer.  `UninitializedNode` is deprecated in the IR (and does not work) so I don't get 90% of the sources.

Unless we can fix one of those, the quality of results using data flow is much worse than the current `StackVariableReachability` version.